crates/shielded_token/src/masp/shielded_wallet.rs

```diff
@@ -462,6 +509,24 @@ pub trait ShieldedQueries<U: ShieldedUtils + MaybeSend + MaybeSync>:
 pub trait ShieldedApi<U: ShieldedUtils + MaybeSend + MaybeSync>:
     ShieldedQueries<U>
 {
+    /// A method that loads the wallet and empties the conversion cache if the
+    /// masp epoch has advanced. This method is only called by functions
+    /// which are relatively fast so the chances of crossing a masp boundary
+    /// are minimal. However, if it happens, the function may produce
+    /// invalid output and need to be run again.
+    #[allow(async_fn_in_trait)]
+    async fn load_with_caching<C: Client + Sync>(
```

Comment:
if you key conversions by address and masp epoch, I think you don't need to worry about clearing the cache. it will grow out of control unless it is trimmed occasionally, though. the trimming could be done when saving or smth